feat(config)!: experimental flags (#28187)
renovatebot/renovate: docs/usage/self-hosted-experimental.md

If set, Renovate will export OpenTelemetry data to the supplied endpoint.
For more information see [the OpenTelemetry docs](opentelemetry.md).

> ## `RENOVATE_PAGINATE_ALL`

Reviewer: Any reason why this is migrated?

Author: It was in the experimental file list so I assumed that it shuold be converted as well.

If set, Renovate will export OpenTelemetry data to the supplied endpoint.
For more information see [the OpenTelemetry docs](opentelemetry.md).

> ## `RENOVATE_CACHE_NPM_MINUTES`

Author: ```suggestion
```

> If multiple `autodiscoverTopics` are used resulting order will be per topic not global.

> > Allowed values:

Author: ```suggestion
Allowed values:
```

The process that runs Renovate must have the correct permissions to delete the config file.

## `RENOVATE_X_DOCKER_HUB_TAGS_DISABLE`

> If set to any value, Renovate will stop using the Docker Hub API (`https://hub.docker.com`) to fetch tags and instead use the normal Docker API for images pulled from `https://index.docker.io`.

Reviewer 2: ```suggestion
By default, Renovate uses the _Docker Hub_ API (`https://hub.docker.com`) to fetch tags.

But when you set `RENOVATE_X_DOCKER_HUB_TAGS_DISABLE` to any value, Renovate will instead fetch tags from the normal _Docker_ API, for images pulled from `https://index.docker.io`.
```

The current text is hard to understand. I had to read multiple times to get the key point about the Docker API and Docker Hub API.

I'm trying to rewrite this so it's easier to understand:

- the default situation
- what changes when you use this config option
- split the sentence into multiple parts
- _emphasize_ the _Docker Hub_ and _Docker_ parts of the relevant sentence, as that is a key difference

Please double check my rewrite before applying it. :wink:

## `RENOVATE_X_DOCKER_MAX_PAGES`

If set to an integer, Renovate will use this as max page number for docker tags lookup on docker registries, instead of the default 20 pages.
This is useful for registries which ignores the `n` parameter in the query string and only return 50 tags per page.

## `RENOVATE_X_EXEC_GPID_HANDLE`

> If set, Renovate will terminate the whole process group of a terminated child process spawned by Renovate.

Reviewer 2: How can you terminate something that's already terminated?

This needs a rewrite to explain more clearly what's going on.

I think you mean something like this:

> When a child process dies, Renovate will kill the parent process as well.

But in programmer words. :upside_down_face:

Author: I think this one's alright.

> ## `RENOVATE_X_USE_OPENPGP`

Reviewer: Any reason why this is migrated?

Author: We will need to pass it via the global config directly, instead of accessing it using GlobalConfig class

Author: Did you mean why it is not converted to a experimental flag?

## `RENOVATE_X_YARN_PROXY`

Configure global Yarn proxy settings if HTTP proxy environment variables are detected.

> # Self-hosted experimental flags

Author: Created a new file for experimental flags. When are done with converting all the env vars to flags this file will contain all the vars except `RENOVATE_X_IGNORE_RE2` and `RENOVATE_X_HARD_EXIT` ([reason for this](https://github.com/renovatebot/renovate/issues/27879#issuecomment-2005459751)).

I wonder if we should keep the `self-hosted-experiemental.md` file even then or somehow include these vars in the new file.
